lib/plugins/features/movies/controller.js

```diff
@@ -1,9 +1,32 @@
 'use strict';
 
 const Movie = require('../../../models/movie');
+const Movies = Movie.collection();
 
 exports.create = async (payload) => {
   const movie = await new Movie().save(payload);
 
   return new Movie({ id: movie.id }).fetch();
 };
+
+exports.find = async (query) => {
+  let movies;
+  if (query.title) {
```

Comment:
You can actually have the logic of checking what query parameters exist inside the `queryBuilder` function, e.g. 

```js
Movies.query((queryBuilder) => {
  if (query.title) {
    queryBuilder.where('title', '=', query.title);
  }
})
.fetch();
```

You can also chain `.where` calls to the `queryBuilder` in separate statements so you don't have to have do a bunch of nested if statements: 

```js
Movies.query((queryBuilder) => {
  if (query.title) {
    queryBuilder.where('title', '=', query.title);
  }

  if (query.date1 && query.date2) {
    queryBuilder.where('release_year', '>=', query.date1);
    queryBuilder.where('release_year', '<=', query.date2);
  }

  ...
})
.fetch();
```